### PR TITLE
feat(classification): refactor IExtractor::extractor

### PR DESCRIPTION
**Pull request by st3iny**
It now takes a message and returns an array of floats.

### CODE REVIEW

File: lib/Service/Classification/FeatureExtraction/CompositeExtractor.php [hunk 1]
```diff
@@ -26,11 +27,13 @@
 namespace OCA\Mail\Service\Classification\FeatureExtraction;
 
 use OCA\Mail\Account;
+use OCA\Mail\Db\Message;
+use function OCA\Mail\array_flat_map;
```
Comment (ChristophWurst) [inline lib/Service/Classification/FeatureExtraction/CompositeExtractor.php:31]: excellent choice

**Review by ChristophWurst — APPROVED** [review]
Smooth

File: lib/Service/Classification/FeatureExtraction/IExtractor.php [hunk 2]
```diff
@@ -45,11 +46,11 @@ public function prepare(Account $account,
...
 
 	/**
-	 * Return the feature value for the given sender address
+	 * Return the feature value for the given message
 	 *
-	 * @param string $email
+	 * @param Message $message
 	 *
-	 * @return float
+	 * @return float[]
```
Comment (ChristophWurst) [inline lib/Service/Classification/FeatureExtraction/IExtractor.php:53]: to take it another step further you could lift this to any kind of primitive rubix allows as input

Comment (st3iny) [reply lib/Service/Classification/FeatureExtraction/IExtractor.php:53]: I like the idea. However, the Rubix DataSet class supports basically anything. We could refactor our code to use the DataSet class more generally, though.

**Pull request merged**